Server/src/tests/controllers/savedGraphsController.spec.ts

```diff
@@ -0,0 +1,295 @@
+          "units": "mm"
+        }
+      ],
+      "id": 1
+    }]
+    mockRequest = {
+      params: {
+        userSavedGraphs: "j.comkj"
+      }
+    }
+    await SavedGraphsController.createRequestForUserSavedGraphs(mockRequest as Request, mockResponse as Response)
+    expect(mockResponse.json).toBeCalledWith(expectedResponse);
+    expect(mockResponse.status).toBeCalledWith(200);
+  });
+
+  test('Valid User Saved Graphs Request; one data set on graph', async () => {
+    let expectedResponse = [{
+      "datasets": [
+        {
+          "id": 1,
+          "color": "red",
+          "shape": "square",
+          "isHidden": false
+        }
+      ],
+      "name": "Test Graph",
+      "axes": [
+        {
+          "variableName": "temperature",
+          "mode": "normal",
+          "zoom": 100,
+          "units": "C"
+        }
+      ],
+      "id": 2
+    }]
+    mockRequest = {
+      params: {
+        userSavedGraphs: "test@t.com"
+      }
+    }
+    await SavedGraphsController.createRequestForUserSavedGraphs(mockRequest as Request, mockResponse as Response)
+    expect(mockResponse.json).toBeCalledWith(expectedResponse);
+    expect(mockResponse.status).toBeCalledWith(200);
+  });
+
+  test('Invalid User Saved Graphs Request', async () => {
+    let expectedResponse = "Invalid user email provided"
+    mockRequest = {
+      params: {
+        userSavedGraphs: "fake@email.com"
+      }
+    }
+    await SavedGraphsController.createRequestForUserSavedGraphs(mockRequest as Request, mockResponse as Response)
+    expect(mockResponse.json).toBeCalledWith(expectedResponse);
+    expect(mockResponse.status).toBeCalledWith(400);
+  });
+
+  test('Valid Graph Insert Request', async () => {
+    let expectedResponse = "Graph successfully saved"
+    mockRequest = {
+      body: {
+        "datasets": [
+          {
+            "id": 100,
+            "color": "blue",
+            "shape": "square",
+            "isHidden": true
+          },
+          {
+            "id": 101,
+            "color": "green",
+            "shape": "circle",
+            "isHidden": false
+          }
+        ],
+        "name": "Adding Graphset Test",
+        "axes": [
+          {
+            "variableName": "density",
+            "mode": "small",
+            "zoom": 100,
+            "units": "cm, C"
+          },
+          {
+            "variableName": "temp",
+            "mode": "big",
+            "zoom": 100,
+            "units": "cm, C"
+          }
+        ],
+        "id": 1,
+        "email": "j.comkj"
+      }
+    }
+    await SavedGraphsController.createRequestForAddingSavedGraph(mockRequest as Request, mockResponse as Response)
+    expect(mockResponse.json).toBeCalledWith(expectedResponse);
+    expect(mockResponse.status).toBeCalledWith(200);
+  });
+
+  test('Invalid Graph Insert Request', async () => {
+    let expectedResponse = "Invalid user email provided"
+    mockRequest = {
+      body: {
+        "datasets": [

```

Comment:
to make unit testing easier we might wanna get a dummy generating lib, so that the objects are dynamically generated and not hardcoded